pkg/tfbridge/schema.go

```diff
@@ -843,7 +883,42 @@ func CoerceTerraformString(schType schema.ValueType, stringValue string) (interf
 	return stringValue, nil
 }
 
-func extractInputsFromOutputs(urn resource.URN, outs resource.PropertyMap,
+func propagateDefaultAnnotations(oldInput, newInput resource.PropertyValue) {
```

Comment:
This code is latent, and will become active once https://github.com/pulumi/pulumi/pull/2531 and an as-of-yet-unpublished PR land.

In essence: once we begin refreshing inputs, doing so will destroy the `__defaults` annotations this PR adds. In order to avoid that, we need to hand `Read` the current inputs as well as the current state. We can then propagate the `__defaults` annotations for properties whose values have not changed.